File: app/src/main/java/com/android/lowes/repository/WeatherRepository.java
package com.android.lowes.repository;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.android.lowes.models.WeatherDetails;
import com.android.lowes.models.WeatherResponse;
import com.android.lowes.requests.WeatherApiClient;

import java.util.List;

public class WeatherRepository {



        private static WeatherRepository instance;
        private WeatherApiClient mWeatherApiCLient;

        public static WeatherRepository getInstance(){
            if(instance == null){
                instance = new WeatherRepository();
            }
            return instance;
        }

        private WeatherRepository() {
            mWeatherApiCLient = WeatherApiClient.getInstance();
        }

//        public LiveData<List<WeatherResponse>> getWeathers(){
//            return mWeatherApiCLient.getWeathers();
//        }

    public LiveData<List<WeatherDetails>> getWeathers(){
            return mWeatherApiCLient.getWeathers();
        }


    public void searchWeathersApi(String query){

        mWeatherApiCLient.searchRecipesApi(query);
    }
    }

File: app/src/main/java/com/android/lowes/repository/WeatherRepository.kt
package com.android.lowes.repository

import androidx.lifecycle.LiveData
import com.android.lowes.models.Weather
import com.android.lowes.models.WeatherDetails
import com.android.lowes.requests.WeatherApiClient

class WeatherRepository private constructor() {
    private val mWeatherApiCLient: WeatherApiClient
    val weathers: LiveData<List<WeatherDetails>?>
        get() = mWeatherApiCLient?.weathers

    fun searchWeathersApi(query: String?) {
        if (query != null) {
            mWeatherApiCLient.searchRecipesApi(query)
        }
    }



    private object HOLDER {
        val INSTANCE = WeatherRepository()
    }

    companion object {
        val instance: WeatherRepository by lazy { HOLDER.INSTANCE }
    }


    init {
        mWeatherApiCLient = WeatherApiClient.instance
    }
}